bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/protocol/philipstv/service/TvPictureService.java

```diff
@@ -0,0 +1,132 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.androidtv.internal.protocol.philipstv.service;
+
+import static org.openhab.binding.androidtv.internal.AndroidTVBindingConstants.*;
+import static org.openhab.binding.androidtv.internal.protocol.philipstv.ConnectionManager.OBJECT_MAPPER;
+import static org.openhab.binding.androidtv.internal.protocol.philipstv.PhilipsTVBindingConstants.*;
+
+import java.io.IOException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.androidtv.internal.protocol.philipstv.ConnectionManager;
+import org.openhab.binding.androidtv.internal.protocol.philipstv.PhilipsTVConnectionManager;
+import org.openhab.binding.androidtv.internal.protocol.philipstv.service.api.PhilipsTVService;
+import org.openhab.binding.androidtv.internal.protocol.philipstv.service.model.TvSettingsUpdateDto;
+import org.openhab.core.library.types.PercentType;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Service for handling commands regarding the TV picture settings
+ *
+ * @author Benjamin Meyer - Initial contribution
+ * @author Ben Rosenblum - Merged into AndroidTV
+ */
+@NonNullByDefault
+public class TvPictureService implements PhilipsTVService {
+
+    private static final int SHARPNESS_NODE_ID = 2131230851;
+    private static final int CONTRAST_NODE_ID = 2131230850;
+    private static final int BRIGHTNESS_NODE_ID = 2131230852;
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    private final PhilipsTVConnectionManager handler;
+
+    private final ConnectionManager connectionManager;
+
+    public TvPictureService(PhilipsTVConnectionManager handler, ConnectionManager connectionManager) {
+        this.handler = handler;
+        this.connectionManager = connectionManager;
+    }
+
+    @Override
+    public void handleCommand(String channel, Command command) {
+        try {
+            if (CHANNEL_BRIGHTNESS.equals(channel) && command instanceof PercentType) {
+                setBrightness(((PercentType) command).intValue());
```

Comment:
I try to stick with Java 11 as my base because of the marketplace.  I maintain a version of this binding for users from 3.2.0 and up.  Trying to implement the Java 17 bits will cause issues for them.

Reply:
> I try to stick with Java 11 as my base because of the marketplace. I maintain a version of this binding for users from 3.2.0 and up. Trying to implement the Java 17 bits will cause issues for them.

I won’t block it, but I suggest to encourage those users to move forward to 4.x. Supporting those older versions is slowly limiting 4.x progress, like these java17 features.

Reply:
Fully agree. I'm cutting it once philips is locked down.  There are just too many 3.x Phillips users to cut that community off.